components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,52 +1,272 @@
+          }
+
+          m_initializer = create_field_initializer<RRTMGPInputsInitializer>();
     }  // RRTMGPRadiation::RRTMGPRadiation
 
     void RRTMGPRadiation::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
 
+        using namespace ekat::units;
+
+        auto kgkg = kg/kg;
+        kgkg.set_string("kg/kg");
+        auto m3 = m * m * m;
+        m3.set_string("m3");
+        auto Wm2 = W / m / m;
+        Wm2.set_string("Wm2");
+        auto nondim = m/m;  // dummy unit for non-dimensional fields
+        auto micron = m / 1000000;
+
+        auto VL  = FieldTag::VerticalLevel;
+        auto COL = FieldTag::Column;
+        auto VAR = FieldTag::Variable;
+        auto CMP = FieldTag::Component;
+        constexpr int NVL = 42;  /* TODO THIS NEEDS TO BE CHANGED TO A CONFIGURABLE */
+
+        auto grid = grids_manager->get_grid("Physics");
+        const int ncol = grid->get_num_local_dofs();
+
+        // Need to hard-code some dimension sizes for now. TODO: find a better way of configuring this
+        int nswbands = 14;
+        int nlwbands = 16;
+        int ngas = 8;
+
+        // Set up dimension layouts
+        FieldLayout scalar2d_layout     { {COL   }, {ncol    } };
+        FieldLayout scalar3d_layout_mid { {COL,VL}, {ncol,NVL} };
+        FieldLayout scalar3d_layout_int { {COL,VL}, {ncol,NVL+1} };
+        // Use VAR field tag for gases for now; consider adding a tag?
+        FieldLayout gas_layout          { {VAR,COL,VL}, {ngas,ncol,NVL} };
+        FieldLayout scalar2d_swband_layout { {CMP,COL}, {nswbands,ncol} };
+
+        // Set required (input) fields here
+        m_required_fields.emplace("pmid" , scalar3d_layout_mid, Pa, grid->name());
+        m_required_fields.emplace("pint", scalar3d_layout_int, Pa, grid->name());
+        m_required_fields.emplace("tmid" , scalar3d_layout_mid, K , grid->name());
+        m_required_fields.emplace("tint" , scalar3d_layout_int, K , grid->name());
+        m_required_fields.emplace("col_dry", scalar3d_layout_mid, kgkg, grid->name());
+        //m_required_fields.emplace("gas_names", gas_names_layout, nondim, grid->name());
+        m_required_fields.emplace("gas_vmr", gas_layout, kgkg, grid->name());
+        m_required_fields.emplace("sfc_alb_dir", scalar2d_swband_layout, nondim, grid->name());
+        m_required_fields.emplace("sfc_alb_dif", scalar2d_swband_layout, nondim, grid->name());
+        m_required_fields.emplace("mu0", scalar2d_layout, nondim, grid->name());
+        m_required_fields.emplace("lwp", scalar3d_layout_mid, kg/m3, grid->name());
+        m_required_fields.emplace("iwp", scalar3d_layout_mid, kg/m3, grid->name());
+        m_required_fields.emplace("rel", scalar3d_layout_mid, micron, grid->name());
+        m_required_fields.emplace("rei", scalar3d_layout_mid, micron, grid->name());
+
+        // Outputs; these needed to be added to both required and computed fields?
```

Comment:
Output can just be "computed".  "required" would be for anything that acts like in/out.  For example, in the p3 interface we designate the `vap_liq_exchange` fields as just "computed" and it works great.

Reply:
One caveat, which @bartgol can correct if I'm wrong, but the initializer can only initialize something that is considered "required".  The idea being that if something is strictly output then it probably doesn't need to be initialized at the beginning of the run.

Reply:
Thanks for catching this. I think this was required before because I was trying to initialize these in the input initializer. Removing that initialization code (since they are computed by radiation and do not need to be input) allowed me to remove them from the "required" list.